Makefile

```diff
@@ -93,10 +93,10 @@ helm-validate-eks-versions:
 	${MAKEFILE_PATH}/test/helm/validate-chart-versions
 
 build-binaries:
-	${MAKEFILE_PATH}/scripts/build-binaries -p ${SUPPORTED_PLATFORMS_LINUX} -v ${VERSION} -d
```

Comment:
The `-d` or `GOPROXY=direct` arguments on the various build commands are important to being able to build locally, when the developer has certain firewall rules in place on their local machine. Is there a way we can keep the no-proxy mode for local builds while discarding it for GitHub Actions builds?

One possible solution is to examine the `CI` environment variable, which [is always set to `true` in GHA](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables). Maybe we can create a makefile variable that contains either `-d` or nothing, depending on whether `CI` is set, and each relevant build command can include that variable.

Reply:
I agree with the idea to improve setting `GOPROXY=direct` for *when the developer has certain firewall rules in place on their local machine*. However, I would think these devs are the minority and we wouldn't want to assume all devs can't access the general Go module repo (`https://proxy.golang.org`). Brainstorming some other ideas:
* add helper method, `set_goproxy()` to scripts that need `-d` / `direct` flag. This would ping `https://proxy.golang.org` and if response is not success, then set to `direct`
* add failure messaging to scripts that _suggests_ setting `GOPROXY=direct` based on certain error messaging
* add makefile local targets, ex: `make build-binaries-local`. either sets `GOPROXY=direct` or passes `-d`